src/rules/banTsIgnoreRule.ts

```diff
@@ -0,0 +1,50 @@
+/**
+ * @license
+ * Copyright 2013 Palantir Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { forEachComment } from "tsutils";
+import * as ts from "typescript";
+import * as Lint from "../index";
+
+export class Rule extends Lint.Rules.AbstractRule {
+    /* tslint:disable:object-literal-sort-keys */
+    public static metadata: Lint.IRuleMetadata = {
+        ruleName: "ban-ts-ignore",
+        description: 'Bans "// @ts-ignore" comments from being used.',
+        optionsDescription: "Not configurable.",
+        options: null,
+        optionExamples: [true],
+        type: "typescript",
+        typescriptOnly: true,
+    };
+    /* tslint:disable:object-literal-sort-keys */
+
+    public static FAILURE_STRING = 'Do not use "// @ts-ignore" comments because it suppresses compilation errors.';
```

Comment:
Grammar nit: Do not use "// @ts-ignore" comments because *they suppress* compilation errors.